app/src/main/java/com/antonioleiva/bandhookkotlin/ui/screens/detail/ArtistActivity.kt

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2016 Alexey Verein
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.antonioleiva.bandhookkotlin.ui.screens.detail
+
+import android.graphics.drawable.BitmapDrawable
+import android.os.Bundle
+import android.support.annotation.VisibleForTesting
+import android.support.design.widget.CollapsingToolbarLayout
+import android.support.design.widget.TabLayout
+import android.support.v4.view.ViewPager
+import android.support.v7.graphics.Palette
+import android.view.MenuItem
+import android.view.View
+import android.view.WindowManager
+import android.widget.ImageView
+import com.antonioleiva.bandhookkotlin.R
+import com.antonioleiva.bandhookkotlin.di.Inject
+import com.antonioleiva.bandhookkotlin.di.Injector
+import com.antonioleiva.bandhookkotlin.ui.activity.BaseActivity
+import com.antonioleiva.bandhookkotlin.ui.adapter.ArtistDetailPagerAdapter
+import com.antonioleiva.bandhookkotlin.ui.entity.ArtistDetail
+import com.antonioleiva.bandhookkotlin.ui.entity.ImageTitle
+import com.antonioleiva.bandhookkotlin.ui.entity.mapper.ArtistDetailDataMapper
+import com.antonioleiva.bandhookkotlin.ui.entity.mapper.ImageTitleDataMapper
+import com.antonioleiva.bandhookkotlin.ui.fragment.AlbumsFragmentContainer
+import com.antonioleiva.bandhookkotlin.ui.presenter.AlbumsPresenter
+import com.antonioleiva.bandhookkotlin.ui.presenter.ArtistPresenter
+import com.antonioleiva.bandhookkotlin.ui.screens.album.AlbumActivity
+import com.antonioleiva.bandhookkotlin.ui.util.getNavigationId
+import com.antonioleiva.bandhookkotlin.ui.util.navigate
+import com.antonioleiva.bandhookkotlin.ui.util.supportsLollipop
+import com.antonioleiva.bandhookkotlin.ui.view.ArtistView
+import com.squareup.picasso.Callback
+import org.jetbrains.anko.find
+
+class ArtistActivity: BaseActivity(), ArtistView, AlbumsFragmentContainer, Injector by Inject.instance {
+
+    override val layoutResource: Int = R.layout.activity_artist
```

Comment:
Maybe don't need to specify this `val` is an `Int`? the rest of vals are not being specified in this class.